lib/irb/cmd/history.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "stringio"
+require_relative "nop"
+require_relative "../pager"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class History < Nop
+      category "IRB"
+      description "Show the input history."
+
+      def execute(*)
+        output = StringIO.new
+        irb_context.io.class::HISTORY.each_with_index.reverse_each do |input, index|
+          header = "#{index}: "
+          header_length = header.size
+
+          lines = input.split("\n")
+          first_line = header + lines[0]
+
+          truncated_input = lines[1..1]
+          truncated_input << "..." if lines.size > 2
+
+          truncated_input = truncated_input.map do |line|
+            " " * header_length + line
+          end
+
+          output.puts [first_line, *truncated_input].join("\n")
```

Comment:
`puts [a, *b].join("\n")` is almost same as `puts(a, *b)`.
When the second line is `''`, `puts ['first-line', ''].join("\n")` will print 1 line but `puts('first-line', '')` will print 2 lines. I think 2 lines are expected in this case.